test_automation/work_order_utility.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import time
+import os
+import sys
+import argparse
+import random
+import json
+import logging
+
+from service_client.generic import GenericServiceClient
+import crypto.crypto as crypto
+import utility.signature as signature
+import worker.worker_details as worker
+from shared_kv.shared_kv_interface import KvStorage
+import utility.utility as enclave_helper
+import utility.file_utils as futils
+from process_request import process_request
+
+logger = logging.getLogger(__name__)
+
+def create_work_order_request(input_json_str1, worker_obj):
+    """ Function to create work order request. Uses input json file and modifies empty parameters in json file. """
+
+    logger.info("------------------Constructing WorkOrderSubmit------------------")
+    work_order_id = ""
+    input_json_temp = json.loads(input_json_str1)
+
+    # compute work order id
+    work_order_id_json = input_json_temp["params"]["workOrderId"]
+    if work_order_id_json == "":
+        work_order_id = hex(random.randint(1, 2**64 -1))
+        input_json_temp["params"]["workOrderId"] = work_order_id
+    else:
+        work_order_id = work_order_id_json
+
+    # compute worker id
+    worker_id_json = input_json_temp["params"]["workerId"]
+    if worker_id_json == "":
+        input_json_temp["params"]["workerId"] = worker_obj.worker_id
+
+    # Convert workloadId to a hex string and update the request
+    workload_id = input_json_temp["params"]["workloadId"]
+    workload_id_hex = workload_id.encode("UTF-8").hex()
+    input_json_temp["params"]["workloadId"] = workload_id_hex
+    input_json_str1 = json.dumps(input_json_temp)
+
+    return input_json_str1, work_order_id
+
+def sign_work_order_request(input_json_str1, worker_obj, sig_obj, private_key):
+    """ Function to sign the work order request. """
+
+    logger.info("------------------Signing WorkOrderSubmit------------------")
+
+    # create session_iv through enclave_helper
+    session_iv = enclave_helper.generate_sessioniv()
+    # create encrypted_session_key from session_iv and worker_encryption_key
+    encrypted_session_key = enclave_helper.generate_encrypted_session_key(session_iv, worker_obj.worker_encryption_key)
+    # sign work order submit request
+    input_json_str1 = sig_obj.generate_client_signature(input_json_str1, worker_obj, private_key, session_iv, encrypted_session_key)
```

Comment:
This line is too long and is hard to read. Please limit lines to 80 characters.

Reply:
Fixed line length to below 80